src/psyclone/transformations.py

```diff
@@ -161,43 +162,39 @@ def _validate(self, node_list):
                         "transformation".format(type(item), self.name))
 
         # Sanity check that we've not been passed the condition part of
-        # an If statement (which is child 0)
-        if isinstance(node_parent, IfBlock):
-            if node_parent.children[0] in node_list:
-                raise TransformationError(
-                    "Cannot apply transformation to the conditional expression"
-                    " (first child) of an If/Case statement. Error in "
-                    "transformation script.")
-
-            # Check that we've not been supplied with both the if and
-            # else clauses of an IfBlock as we can't put them both in
-            # a region without their parent.
-            if len(node_list) > 1:
-                raise TransformationError(
-                    "Cannot enclose both the if- and else- clauses of an "
-                    "IfBlock by a {0} transformation. Apply the "
-                    "transformation to the IfBlock node instead.".
-                    format(self.name))
+        # an If statement or the bounds of a Loop. If the first node in the
+        # list is a Schedule then that must be the only entry.
+        from psyclone.profiler import ProfileNode
+        from psyclone.extractor import ExtractNode
+        if isinstance(node_parent, IfBlock) and \
```

Comment:
Is there a more generic test here rather than testing for specific if and loop clauses i.e. that if one of the nodes in the list is a schedule then it must be the only node in the list?

Reply:
Yes, I've improved it slightly although I then still need to check for the case when the parent is an IfBlock or Loop. Once everything has a Schedule this can be improved - I've added a TODO referring to #542.